controller/MiNiFiController.cpp

```diff
@@ -171,84 +168,72 @@ int main(int argc, char **argv) {
     if (result.count("stop") > 0) {
       auto& components = result["stop"].as<std::vector<std::string>>();
       for (const auto& component : components) {
-        auto socket = secure_context != nullptr ? stream_factory_->createSecureSocket(host, port, secure_context) : stream_factory_->createSocket(host, port);
-        if (minifi::controller::stopComponent(std::move(socket), component))
+        if (minifi::controller::stopComponent(socket_data, component))
           std::cout << component << " requested to stop" << std::endl;
         else
-          std::cout << "Could not connect to remote host " << host << ":" << port << std::endl;
+          std::cout << "Could not connect to remote host " << socket_data.host << ":" << socket_data.port << std::endl;
       }
     }
 
     if (result.count("start") > 0) {
       auto& components = result["start"].as<std::vector<std::string>>();
       for (const auto& component : components) {
-        auto socket = secure_context != nullptr ? stream_factory_->createSecureSocket(host, port, secure_context) : stream_factory_->createSocket(host, port);
-        if (minifi::controller::startComponent(std::move(socket), component))
+        if (minifi::controller::startComponent(socket_data, component))
           std::cout << component << " requested to start" << std::endl;
         else
-          std::cout << "Could not connect to remote host " << host << ":" << port << std::endl;
+          std::cout << "Could not connect to remote host " << socket_data.host << ":" << socket_data.port << std::endl;
       }
     }
 
     if (result.count("c") > 0) {
       auto& components = result["c"].as<std::vector<std::string>>();
       for (const auto& connection : components) {
-        auto socket = secure_context != nullptr ? stream_factory_->createSecureSocket(host, port, secure_context)
-                                                : stream_factory_->createSocket(host, port);
-        if (minifi::controller::clearConnection(std::move(socket), connection)) {
+        if (minifi::controller::clearConnection(socket_data, connection)) {
           std::cout << "Sent clear command to " << connection << ". Size before clear operation sent: " << std::endl;
-          socket = secure_context != nullptr ? stream_factory_->createSecureSocket(host, port, secure_context)
-                                             : stream_factory_->createSocket(host, port);
-          if (minifi::controller::getConnectionSize(std::move(socket), std::cout, connection) < 0)
-            std::cout << "Could not connect to remote host " << host << ":" << port << std::endl;
+          if (!minifi::controller::getConnectionSize(socket_data, std::cout, connection))
+            std::cout << "Could not connect to remote host " << socket_data.host << ":" << socket_data.port << std::endl;
```

Comment:
Good point, I think it was supposed to give information of the current connection size before clearing it, but it is done after the command is sent and processed so it will always show 0. I think it's unnecessary so I removed it in 45c677990eeb0cb223814b3c55d37e84711a9b03